app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/model/event/EventTest.kt

```diff
@@ -125,4 +128,106 @@ class EventTest {
 
     assertNull(updatedEvent.supplies[supply.id]!!.assignedTo[account2.firebaseAuthUID])
   }
+
+  @Test
+  fun pollsTest() {
+    val poll =
+        ChimpagnePoll(
+            title = "title",
+            query = "is this app for event organisation ?",
+            options =
+                mapOf(
+                    Pair("1", ChimpagnePollOption("1", "yes")),
+                    Pair("2", ChimpagnePollOption("2", "no"))),
+            votes = emptyMap())
+
+    var loading = true
+    eventManager.atomic.createPoll(event.id, poll, { loading = false }, { assertTrue(false) })
+    while (loading) {}
+    Thread.sleep(SLEEP_AMOUNT_MILLIS)
+    var updatedEvent = ChimpagneEvent()
+    loading = true
+    eventManager.getEventById(
+        event.id,
+        {
+          updatedEvent = it!!
+          loading = false
+        },
+        { assertTrue(false) })
+    while (loading) {}
+    Thread.sleep(SLEEP_AMOUNT_MILLIS)
+
+    assertEquals(poll.id, updatedEvent.polls[poll.id]!!.id)
+
+    loading = true
+    eventManager.atomic.castPollVote(
+        event.id,
+        poll.id,
+        account0.firebaseAuthUID,
+        "1",
+        { loading = false },
+        { assertTrue(false) })
+    while (loading) {}
+
+    loading = true
+    eventManager.atomic.castPollVote(
+        event.id,
+        poll.id,
+        account1.firebaseAuthUID,
+        "1",
+        { loading = false },
+        { assertTrue(false) })
+    while (loading) {}
+
+    loading = true
+    eventManager.atomic.castPollVote(
+        event.id,
+        poll.id,
+        account2.firebaseAuthUID,
+        "2",
+        { loading = false },
+        { assertTrue(false) })
+    while (loading) {}
```

Comment:
a helping function waitLoading would reduce copy pasted code here

Reply:
This cannot be done for database related functions as established.
Plus you would be replacing 1 line of code by another 1 line of code in the tests, so I do not see the point of trying to make modifications.

Reply:
> a helping function waitLoading would reduce copy pasted code here

It's hard to do has you cannot directly pass a pointer in Kotlin. You need to wrap loading into an object... Not sure we want to make our testing framework more complex